opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -124,15 +123,24 @@ message LabelValue {
 
 // A timestamped measurement.
 message Point {
-  // Must be present for counter/cumulative metrics. The time when the
-  // cumulative value was reset to zero. The cumulative value is over the time
-  // interval (start_timestamp, timestamp]. If not specified, the backend can
-  // use the previous recorded value.
-  google.protobuf.Timestamp start_timestamp = 1;
-
-  // The moment when this point was recorded.
-  // If not specified, the timestamp will be decided by the backend.
-  google.protobuf.Timestamp timestamp = 2;
+  // start_time_unixnano is the time when the cumulative value was reset to zero.
+  // This is used for Counter type only. For Gauge the value is not specified and
+  // defaults to 0.
+  //
+  // The cumulative value is over the time interval [start_time_unixnano, timestamp_unixnano].
```

Comment:
For the first data point after reset it can be that `start_timestamp==timestamp`. This is a valid situation and must be representable by our model. However if we follow your suggestion we have a problem because for any value of `a` intervals `[a,a)` or `(a,a]` cannot exist (you cannot both include `a` in interval and exclude it at the same time.

Closed interval [start_time_unixnano, timestamp_unixnano] sounds right to me here. I do not see how it prevents from representing delta over time.

Reply:
Start time will be the previous end time, so you will always send (t0, t1] (t1, t2]

Reply:
So you are saying it is not valid to have a message where `start_time_unixnano==timestamp_unixnano`. I do not see why we need such a limitation, it seems a normal situation to me. When I just start counting and send the very first value immediately (the counter can be 0 or can be non-zero in which can it may make sense to send it).

I still do not understand what is the problem that you are seeing with closed interval.

Reply:
For the close intervals:
[t0, t1] [t1, t2]

I cannot calculate the value for [t0, t1] (or cannot do any merge of the intervals) because I may double count a value, so I cannot compute the cumulative value.

Reply:
In your example why is there a requirement that one interval's last timestamp is equal to the next intervals start timestamp? I would expect the intervals to look like this:
[t0, t1] [t2, t3], where t2>t1 and t1 is the last reported timestamp (which does not need to be the time when reset happens.
Perhaps there is such requirement and the SDK is expected to work like this, please clarify.

Reply:
See here as well https://github.com/OpenObservability/OpenMetrics/pull/126/files (line 110).

So start_time is used by some backends to determine restarts (especially in case of push mechanism), by not having the same start/end between deltas these systems may guess that a restart happens between these points. It is not a requirement from the protocol to have these values to be the same, but in order to be able to send proper deltas for these backends we need to not have gaps in the interval.

Reply:
OK, in that case we need to use `(start_timestamp, timestamp]` (the opposite is meaningless because it excludes the timestamp that is reported by this data point).

The consequence of this requirement is that you cannot send a data point immediately after restart such that numerically the nanosecond timestamp is equal to the nanosecond timestamp of the moment when reset is performed.

You can however send a data point immediately before the reset and timestamp both the data point and reset even with same numerical nanosecond value. This sounds like what you would also want to do in the code if your metric value fetching code is also responsible for resetting logic, so I think this is a reasonable approach.

I will change to `(start_timestamp, timestamp]`

Reply:
Done.